common/src/main/java/com/gentics/mesh/OptionsLoader.java

```diff
@@ -53,11 +54,19 @@ public static MeshOptions createOrloadOptions(MeshOptions defaultOption, String.
 		MeshOptions options = loadMeshOptions(defaultOption);
 		applyNonYamlProperties(defaultOption, options);
 		applyEnvironmentVariables(options);
+		applyReplacements(options);
 		applyCommandLineArgs(options, args);
 		options.validate();
 		return options;
 	}
 
+	private static void applyReplacements(MeshOptions options) {
+		String nodeName = options.getNodeName();
+		if ("$HOSTNAME".equals(nodeName)) {
```

Comment:
I think replacing the variable instead would be more intuitive. This is how it works everywhere else.